src/CSV.jl

```diff
@@ -96,6 +101,17 @@ function checkvalidsource(source)
         throw(ArgumentError("\"$source\" is not a valid file"))
 end
 
+function allocate(rowsguess, ncols, typecodes)
+    tapes = Vector{UInt64}[Mmap.mmap(Vector{UInt64}, usermissing(typecodes[i]) ? 0 : rowsguess) for i = 1:ncols]
```

Comment:
Changes here include:
  * if a user passes `Missing` as a column type, then we avoid allocating any tape/poslen vectors at all; any value found in the cell while parsing will still be parsed, but no value/position/length will be stored anywhere. The `CSV.Column{Missing}` just checks that `getindex` is in bounds and then returns `missing`. This can help a lot for large files that have entire columns of null values (though you have to pass `Missing` as the column type manually). It also provides a way we could do really efficient "baked in" column selection while parsing; we would just set the non-selected columns as having column type `Missing` and they essentially get skipped
  * The other change here is that if the user provides the column type for any other kind of column, we don't allocate the `poslen` vector. The `poslen` vector for a column tracks the position and length of each cell, in case we need to promote the column to a string at any point later on while parsing. For user-provided types, the user is essentially declaring the column type should be X, so we don't need to track this additional info (which costs memory). Previously, the column `tape` stored the raw parsed values and poslen values in the same array; separating them into `tape` and `poslen` allows us to avoid the `poslen` when it's not needed.